src/components/radio/Radio.tsx

```diff
@@ -0,0 +1,61 @@
+import { Component, createEffect, createSignal } from 'solid-js'
+
+import StyledIcon from './StyledIcon'
+import StyledInput from './StyledInput'
+import StyledLabel from './StyledRadio'
+
+export type RadioProps = {
+  label?: JSX.Element | string
+  labelPosition?: 'right' | 'left'
+  checked?: boolean
+  indeterminate?: boolean
+  disabled?: boolean
+  className?: string
+  onChange?: (checked: boolean) => void
+}
+
+const Radio: Component<RadioProps> = (props) => {
+  const [isChecked, setIsChecked] = createSignal(false)
+
+  createEffect(() => {
+    if (props.checked !== null && props.checked !== undefined) {
```

Comment:
A nice tip is `props.checked != null` which is essentially the same as `props.checked !== null && props.checked !== undefined`

Reply:
(no need to update, just giving as a tip)

Reply:
Thanks for the tip :)